blueocean-github-pipeline/src/main/java/io/jenkins/blueocean/blueocean_github_pipeline/GithubScmFileContent.java

```diff
@@ -0,0 +1,139 @@
+package io.jenkins.blueocean.blueocean_github_pipeline;
+
+import org.kohsuke.stapler.export.Exported;
+import org.kohsuke.stapler.export.ExportedBean;
+
+/**
+ * @author Vivek Pandey
+ */
+@ExportedBean
+public class GithubScmFileContent{
```

Comment:
Ah, thats already taken care at `ScmContentProvider.saveContent()` level. For every provider, like gitlab, you have gitlab specific saveContent() implemented and it knows what to expect from  read and respond with gitlab specific save content response. Tomorrow if we have a totally different scm whose request/response looks very different from GitHub that will work too. I am assuming UI bit is implemented in purely extensible way as well to allow different implementations to load/save in fairly pluggable way.